frontend/src/pages/modelServing/screens/global/DeleteInferenceServiceModal.tsx

```diff
@@ -64,6 +72,12 @@ const DeleteInferenceServiceModal: React.FC<DeleteInferenceServiceModalProps> =
             ...(isKServeNIMEnabled && pvcName
               ? [deletePvc(pvcName, inferenceService.metadata.namespace)]
               : []),
+            ...(isKServeNIMEnabled && project
+              ? [
+                  deleteSecret(project.metadata.name, nimSecretName),
+                  deleteSecret(project.metadata.name, imagePullSecretName),
```

Comment:
Nit. if you don't find these names, you'll fire off requests to no effect -- probably could write this logic a little more defensive and build out something to spread into the `Promise.all` before based on the various conditions